Source/Core/DiscIO/FileSystemGCWii.cpp

```diff
@@ -20,131 +22,393 @@
+{
+  if (!IsValid())
+    return nullptr;
+
+  // Build a cache (unless there already is one)
+  if (m_offset_file_info_cache.empty())
+  {
+    u32 fst_entries = m_root.GetSize();
+    for (u32 i = 0; i < fst_entries; i++)
+    {
+      CFileInfoGCWii file_info(m_root, i);
+      if (!file_info.IsDirectory())
+        m_offset_file_info_cache.emplace(file_info.GetOffset() + file_info.GetSize(), i);
+    }
+  }
+
+  // Get the first file that ends after disc_offset
+  auto it = m_offset_file_info_cache.upper_bound(disc_offset);
+  if (it == m_offset_file_info_cache.end())
+    return nullptr;
+  std::unique_ptr<IFileInfo> result(std::make_unique<CFileInfoGCWii>(m_root, it->second));
+
+  // If the file's start isn't after disc_offset, success
+  if (result->GetOffset() <= disc_offset)
+    return result;
+
+  return nullptr;
+}
+
+u64 CFileSystemGCWii::ReadFile(const IFileInfo* file_info, u8* buffer, u64 max_buffer_size,
+                               u64 offset_in_file) const
+{
+  if (!file_info || file_info->IsDirectory())
     return 0;
 
-  if (_OffsetInFile >= pFileInfo->m_FileSize)
+  if (offset_in_file >= file_info->GetSize())
     return 0;
 
-  u64 read_length = std::min(_MaxBufferSize, pFileInfo->m_FileSize - _OffsetInFile);
+  u64 read_length = std::min(max_buffer_size, file_info->GetSize() - offset_in_file);
 
   DEBUG_LOG(DISCIO, "Reading %" PRIx64 " bytes at %" PRIx64 " from file %s. Offset: %" PRIx64
                     " Size: %" PRIx64,
-            read_length, _OffsetInFile, _rFullPath.c_str(), pFileInfo->m_Offset,
-            pFileInfo->m_FileSize);
+            read_length, offset_in_file, file_info->GetPath().c_str(), file_info->GetOffset(),
+            file_info->GetSize());
 
-  m_rVolume->Read(pFileInfo->m_Offset + _OffsetInFile, read_length, _pBuffer, m_Wii);
+  m_volume->Read(file_info->GetOffset() + offset_in_file, read_length, buffer, m_wii);
   return read_length;
 }
 
-bool CFileSystemGCWii::ExportFile(const std::string& _rFullPath,
-                                  const std::string& _rExportFilename)
+bool CFileSystemGCWii::ExportFile(const IFileInfo* file_info,
+                                  const std::string& export_filename) const
 {
-  if (!m_Initialized)
-    InitFileSystem();
-
-  const SFileInfo* pFileInfo = FindFileInfo(_rFullPath);
-
-  if (!pFileInfo)
+  if (!file_info || file_info->IsDirectory())
     return false;
 
-  u64 remainingSize = pFileInfo->m_FileSize;
-  u64 fileOffset = pFileInfo->m_Offset;
+  u64 remaining_size = file_info->GetSize();
+  u64 file_offset = file_info->GetOffset();
 
-  File::IOFile f(_rExportFilename, "wb");
+  File::IOFile f(export_filename, "wb");
   if (!f)
     return false;
 
   bool result = true;
 
-  while (remainingSize)
+  while (remaining_size)
   {
     // Limit read size to 128 MB
-    size_t readSize = (size_t)std::min(remainingSize, (u64)0x08000000);
+    size_t read_size = (size_t)std::min(remaining_size, (u64)0x08000000);
```

Comment:
The only reason I'm touching this part of the code is to change the variable naming style throughout all of the filesystem code (as suggested by delroth). I'd prefer to not need to make any additional changes just because of the renaming commit - this PR is already big enough as it is.